test/FunctionalTests/E2ETestBase.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EntropyTests
+{
+    public abstract class E2ETestBase
+    {
+        private ITestOutputHelper _output;
+
+        protected E2ETestBase(ITestOutputHelper output, string siteName, int basePort)
+        {
+            _output = output;
+            SiteName = siteName;
+            BasePort = basePort;
+        }
+
+        public string SiteName { get; }
+
+        public int BasePort { get; }
+
+#if NETCOREAPP1_1
+        [Fact]
+        public Task KestrelX64CoreCLR()
+        {
+            return RunTestAsync(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, BasePort);
+        }
+
+        [ConditionalFact(Skip = "Skipped until https://github.com/aspnet/Hosting/issues/949")]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task KestrelX86CoreCLRWindows()
+        {
+            return RunTestAsync(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, BasePort + 1);
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task KestrelX64CLRWindows()
+        {
+            return RunTestAsync(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, BasePort + 2);
+        }
+
+        [ConditionalFact(Skip = "Skipped until https://github.com/aspnet/Hosting/issues/949")]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task KestrelX86CLRWindows()
+        {
+            return RunTestAsync(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x86, BasePort + 3);
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task WebListenerX64CoreCLRWindows()
+        {
+            return RunTestAsync(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, BasePort + 4);
+        }
+
+        [ConditionalFact(Skip = "Skipped until https://github.com/aspnet/Hosting/issues/949")]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task WebListenerX86CoreCLRWindows()
```

Comment:
@Tratcher should the weblistener variants just go away? Alternatively what should we do to fix it - change the project? #188